src/model/clients/__tests__/HttpClient.js

```diff
@@ -0,0 +1,336 @@
+      timeout: 30000,
+      headers: {
+        TestHeader: 'test value',
+        TestHeader2: 'test value',
+        TestHeader3: 'test value',
+      },
+      data: {
+        test: 'test',
+      },
+    };
+    const result = await HttpClient.put('/api/test', config);
+
+    expect(result).toStrictEqual({
+      testResponse: 'test value',
+    });
+  });
+
+  it('executes PATCH requests with the shorthand command', async () => {
+    nock('https://httpclient.com')
+      .patch('/api/test', {
+        test: 'test',
+      })
+      // eslint-disable-next-line no-magic-numbers
+      .reply(200, {
+        testResponse: 'test value',
+      });
+
+    const config = {
+      baseURL: 'https://httpclient.com',
+      timeout: 30000,
+      headers: {
+        TestHeader: 'test value',
+        TestHeader2: 'test value',
+        TestHeader3: 'test value',
+      },
+      data: {
+        test: 'test',
+      },
+    };
+    const result = await HttpClient.patch('/api/test', config);
+
+    expect(result).toStrictEqual({
+      testResponse: 'test value',
+    });
+  });
+
+  it('executes DELETE requests with the shorthand command', async () => {
+    nock('https://httpclient.com')
+      .delete('/api/test')
+      // eslint-disable-next-line no-magic-numbers
+      .reply(200, {
+        testResponse: 'test value',
+      });
+
+    const config = {
+      baseURL: 'https://httpclient.com',
+      timeout: 30000,
+      headers: {
+        TestHeader: 'test value',
+        TestHeader2: 'test value',
+        TestHeader3: 'test value',
+      },
+    };
+    const result = await HttpClient.delete('/api/test', config);
+
+    expect(result).toStrictEqual({
+      testResponse: 'test value',
+    });
+  });
+
+  it('provides the response body as an error if the request fails', async () => {
+    nock('https://httpclient.com')
+      .get('/api/test')
+      // eslint-disable-next-line no-magic-numbers
+      .reply(404, {
+        testResponse: 'test value',
+      });
+
+    const config = {
+      baseURL: 'https://httpclient.com',
+      timeout: 30000,
+      headers: {
+        TestHeader: 'test value',
+        TestHeader2: 'test value',
+        TestHeader3: 'test value',
+      },
+    };
+
+    try {
+      await HttpClient.get('/api/test', config);
+    } catch (err) {
+      expect(err).toStrictEqual({
+        testResponse: 'test value',
+      });
+    }
+  });
+
+  it('displays an error message if the request is misconfigured', async () => {
+    const config = {
+      baseURL: 'https://httpclient.com',
+      timeout: 30000,
+      headers: {
+        TestHeader: 'test value',
+        TestHeader2: 'test value',
+        TestHeader3: 'test value',
+      },
+    };
+
+    try {
+      await HttpClient.get({ url: '/api/test' }, config);
+    } catch (err) {
+      expect(err).toStrictEqual(
+        `This is embarrassing, we couldn't execute this request. Please try again in a few moments.`
```

Comment:
what's the error code in this case? is it a 404 or something like that? Or is it that there is no server at all to talk to?

Reply:
There's no code here, this is just if you set up the client the wrong way

```javascript
// Wrong
await HttpClient.get({ url: '/api/test' }, config);

// Correct
await HttpClient.get('/api/test', config);
```